### ISSUE TITLE
Apply jitter or rate-limit to PR submissions

### ISSUE THREAD

**Issue by vmarkovtsev** [dependabot/dependabot-core#3822]
I've configured dependabot to receive weekly version bumps. The unit tests in my GHA-s query Auth0 and depend on their OAuth client ID and secret. When dependabot submits ~10 PRs at the same time, my access to Auth0 gets rate-limited and the unit tests fail.

I request the ability to randomize the PRs submission times with some jitter of the configurable amplitude, much like Kubernetes does with pod (re)starts. Alternatively, the ability to throttle the PRs so that no more than X are running Actions at the same time.

**Comment by jeffwidman** [comment]
👋 Sorry for the slow response.

The problem makes sense, but the architecture of Dependabot is such that it runs a "job" for a given ecosystem and then from there fires off PR's for all the package updates it finds. TBH, we're very unlikely to implement a slowdown in firing off those PR's in the Dependabot service that runs natively on GitHub. Among other things, one problem this creates is that you may want to merge the PR's in a certain order, and it just happened the PR you need to merge first is the one it opens last. It also creates complexities in our internal infra because we have to add delayed jobs rather than just firing them off to a queue. Could this be done? Yes. But does it create enough value for enough users to be worth the complexity in both configuration and implementation? Probably not.

However, the root problem of hitting rate-limits in CI might be solvable a few other ways:
* If Dependabot created PR's as soon as the new versions were released, that would inherently use jitter:
  * https://github.com/dependabot/dependabot-core/issues/3488
* If Dependabot opened a single PR for multiple deps, that would result in fewer CI runs:
  * #1190
* You could add rate-limiting or jitter manually in your CI system. Here's a good writeup from someone who did just that: https://www.cazzulino.com/github-actions-rate-limiting.html

I'm going to close as wontfix, but hopefully one of the above solutions will work for you.
